WinUIGallery/DataModel/ControlInfoDataSource.cs

```diff
@@ -194,105 +197,70 @@ public async Task<ControlInfoDataGroup> GetGroupFromItemAsync(string uniqueId)
 
         private async Task GetControlInfoDataAsync()
         {
-            lock (_lock)
+            if (this.Groups.Count() != 0)
             {
-                if (this.Groups.Count() != 0)
-                {
-                    return;
-                }
+                return;
             }
 
-            string jsonText = await FileLoader.LoadText("DataModel/ControlInfoData.json");
-
-            JsonObject jsonObject = JsonObject.Parse(jsonText);
-            JsonArray jsonArray = jsonObject["Groups"].GetArray();
-
-            lock (_lock)
+            var jsonFile = await FileLoader.GetJsonFile("DataModel/ControlInfoData.json");
+            using FileStream openStream = File.OpenRead(jsonFile.Path);
+            var controlInfoDataGroup = await JsonSerializer.DeserializeAsync<Root>(openStream, new JsonSerializerOptions
```

Comment:
Why use a stream and not just the text?

Reply:
To use Async we need to use Stream. Text is not part of the input parameters in `DeserializeAsync` method, If there is no need for Async, we can use Text in `Deserialize` method.
so should we keep async or sync?

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to?pivots=dotnet-7-0

Reply:
There really is no need to make the parsing be async. The app cannot start when we don't have this information so we need to wait for this. And the fact that both DeserializeAsync and the function calling DeserializeAsync are both awaited indicates we can take use of the synchronous function.

Reply:
done, i changed async to sync and we are using Text now.